Volume/Supporting/Notifications.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  Notifications.swift
+//  Volume
+//
+//  Created by Hanzheng Li on 11/17/21.
+//  Copyright © 2021 Cornell AppDev. All rights reserved.
+//
+
+import AppDevAnalytics
+import Foundation
+import UserNotifications
+
+class Notifications: NSObject {
+    static let shared = Notifications()
+    let center = UNUserNotificationCenter.current()
+    
+    private override init() {
+        super.init()
+        center.delegate = self
+    }
+    
+    func requestAuthorization() {
+        center.getNotificationSettings { settings in
+            if settings.authorizationStatus == .notDetermined {
+                self.center.requestAuthorization(options: [.alert, .badge, .sound]) { granted, error in
+                    if granted {
+                        let event = VolumeEvent.enableNotification.toEvent(.notification, value: "", navigationSource: .unspecified)
+                        AppDevAnalytics.log(event)
+                    } else if let error = error {
```

Comment:
Should we handle this better? Maybe notify user they can go to settings later if they change their mind?

Reply:
Hmmm. This seems like a higher-fidelity feature that requires some communication with design to implement. Will look into this